backends-velox/src/main/scala/org/apache/gluten/execution/GenerateExecTransformer.scala

```diff
@@ -146,20 +147,48 @@ object PullOutGenerateProjectHelper extends PullOutProjectHelper {
             // generator.child is other expression, e.g Literal/CreateArray/CreateMap
             expressionMap.values.head
           }
-          Seq(newGeneratorChild)
+          val newGeneratorChildren = Seq(newGeneratorChild)
+
+          // Avoid using elimainateProjectList to create the project list
+          // because newGeneratorChild can be a duplicated Attribute in generate.child.output.
+          // The native side identifies the last field of projection as generator's input.
+          generate.copy(
+            generator =
+              generate.generator.withNewChildren(newGeneratorChildren).asInstanceOf[Generator],
+            child = ProjectExec(generate.child.output ++ newGeneratorChildren, generate.child)
+          )
+        case JsonTuple(Seq(jsonObj, jsonPaths @ _*)) =>
+          val getJsons: IndexedSeq[Expression] = {
+            jsonPaths.map {
+              case jsonPath if jsonPath.foldable =>
+                Option(jsonPath.eval()) match {
+                  case Some(path) =>
+                    GetJsonObject(jsonObj, Literal.create(JSON_PATH_PREFIX + path))
+                  case _ =>
+                    Literal.create(null)
+                }
+              case jsonPath =>
+                GetJsonObject(jsonObj, Concat(Seq(Literal.create(JSON_PATH_PREFIX), jsonPath)))
```

Comment:
The prefix is just for adapting to GetJsonObject. Maybe, we can remove this handling in the future by making path without "$." recognized. It would be good to leave a "TODO" comment here.